airflow/logging/airflow_task_handler.py

```diff
@@ -0,0 +1,244 @@
+                    except (AirflowConfigException, ValueError):
+                        pass
+
+                    response = requests.get(url, timeout=timeout)
+                    response.raise_for_status()
+                    log += '\n' + response.text
+                except Exception:
+                    log += "*** Failed to fetch log file from worker.\n".format(
+                        **locals())
+        return log
+
+    def _get_directory(self, ti):
+        '''
+        Get given task instance's log file directory
+        :param ti: task instance object
+        '''
+        return os.path.join(ti.dag_id, ti.task_id)
+
+    def _get_filename(self, ti):
+        '''
+        Get relative log filename for given task instance.
+        :param ti: task instance object
+        '''
+        directory = self._get_directory(ti)
+        filename = "{}.log".format(ti.execution_date.isoformat())
+        return os.path.join(directory, filename)
+
+    def _init_file(self, task_instance):
+        '''
+        Create log directory and give it correct permissions.
+        :param task_instance: task instance object
+        :return full name of the file
+        '''
+        # To handle log writing when tasks are impersonated, the log files need to
+        # be writable by the user that runs the Airflow command and the user
+        # that is impersonated. This is mainly to handle corner cases with the
+        # SubDagOperator. When the SubDagOperator is run, all of the operators
+        # run under the impersonated user and create appropriate log files
+        # as the impersonated user. However, if the user manually runs tasks
+        # of the SubDagOperator through the UI, then the log files are created
+        # by the user that runs the Airflow command. For example, the Airflow
+        # run command may be run by the `airflow_sudoable` user, but the Airflow
+        # tasks may be run by the `airflow` user. If the log files are not
+        # writable by both users, then it's possible that re-running a task
+        # via the UI (or vice versa) results in a permission error as the task
+        # tries to write to a log file created by the other user.
+        log_base = os.path.expanduser(conf.get('core', 'BASE_LOG_FOLDER'))
+        directory = os.path.join(log_base, self._get_directory(task_instance))
+        # Create the log file and give it group writable permissions
+        # TODO(aoen): Make log dirs and logs globally readable for now since the SubDag
+        # operator is not compatible with impersonation (e.g. if a Celery executor is used
+        # for a SubDag operator and the SubDag operator has a different owner than the
+        # parent DAG)
+        if not os.path.exists(directory):
+            # Create the directory as globally writable using custom mkdirs
+            # as os.makedirs doesn't set mode properly.
+            mkdirs(directory, 0o775)
```

Comment:
I though we fixed this already? Please make sure to use proper umask and don't just assume permissions.